### PR TITLE
3 get api categories

### PR DESCRIPTION
**Pull request by rose-phillips**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const express = require("express");
+const { getCategories } = require("./controllers/games-controller");
+const app = express();
+
+app.use(express.json());
```
Comment (Geo-Beard) [inline app.js]: Don't need to use this yet - remove for now then add back in in future ticket when it's needed.

File: app.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+app.get("/api/categories", getCategories);
+
+// error handling
+app.get("/*", (req, res) => {
+  res.status(404).send({ msg: "invalid path" });
+});
+
+app.use((err, req, res, next) => {
+  console.log(err, "unhandled error");
```
Comment (Geo-Beard) [inline app.js]: Try to make sure you don't have any console.log()'s left in your code. Remove :)

File: listen.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+const app = require("./app.js");
+
+app.listen(8080, (err) => {
```
Comment (Geo-Beard) [inline listen.js]: Don't think you need this app.listen() - only useful if you're using Insomnia/some other manual endpoint testing. Supertest is acting as the client for you.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    test("200 status", () => {
+      return request(app).get("/api/categories").expect(200);
+    });
+    test("get array of category objects with slug and description properties", () => {
+      return request(app)
+        .get("/api/categories")
+        .expect(200)
+        .then((response) => {
+          expect(response.body.categories).toEqual(expect.any(Array));
+          expect(Object.keys(response.body.categories[0])).toEqual(
```
Comment (Geo-Beard) [inline __tests__/app.test.js]: Think about using a .forEach() to check the full shape of the array. Also add a test to check the array length to make sure you're getting the full array that we expect.

**Comment by Geo-Beard** [comment]
Added some comments :) If you make those changes then you should be good to merge and move on. If you'd like me to check it again that's fine too! I'll DM.

File: listen.js (hunk 3 above)
Comment (rose-phillips) [reply listen.js]: can I delete the whole listen file?

**Pull request merged**